app/controllers/bookings_controller.rb

```diff
@@ -0,0 +1,44 @@
+class BookingsController < ApplicationController
+  before_action :set_event
+  before_action :set_booking, only: %i[update cancelation]
+
+  def create
+    @booking = @event.bookings.new(booking_params)
+
+    if @booking.save
+      redirect_to root_path
+    else
+      render "pages/home", status: :unprocessable_entity
+    end
+  end
+
+  def update
+    if @booking.update(booking_params)
+      redirect_to root_path
+    else
+      render "pages/home", status: :unprocessable_entity
+    end
+  end
+
+  def cancelation
+    if @booking.canceled?
+      @booking.update(status: "canceled")
+    else
+      render "pages/home", status: :unprocessable_entity
+    end
+  end
```

Comment:
You don't need an if else statement here.
Just cancel the booking and the button to cancel will not be available if the booking is already canceled

Reply:
```
  def cancel_booking
    if @booking.canceled?
      @booking.update(status: "canceled")
    end
    render "pages/home", status: :unprocessable_entity
  end
```

Reply:
Again you don't need an if statement.
If the booking is not canceled yet you cancel it, if the booking is already canceled you can update it too and it won't change it 🤷

Reply:
I see what you mean now, in my head was that the user creates a new booking if he/her cancelled the booking
so, without an if statement and to be able to toggle between the 'canceled' and 'not canceled' status I propose this:
```
def cancel_booking
  @booking.update(status: !@booking.canceled?)
  render "pages/home", status: :unprocessable_entity
end
```

Reply:
We don't want to toggle. A user should not be able to uncancel a booking.
When the booking is not canceled we display a cancel button which triggers this action to cancel the booking
When the booking is canceled the button does not appear anymore so he can't call this action if it's already canceled